packages/devtools_shared/lib/src/memory/heap_sample.dart

```diff
@@ -53,9 +53,9 @@ class HeapSample {
         'used': used,
         'external': external,
         'gc': isGC,
-        'adb_memoryInfo': adbMemoryInfo.toJson(),
-        'memory_eventInfo': memoryEventInfo.toJson(),
-        'raster_cache': rasterCache.toJson(),
+        'adb_memoryInfo': adbMemoryInfo,
+        'memory_eventInfo': memoryEventInfo,
+        'raster_cache': rasterCache,
```

Comment:
I recommend that we don't make unnecessary changes to devtools_shared that aren't test covered. I don't think this code has good test coverage to verify nothing is broken here.

Reply:
Added test coverage. Heap sample is part of chart data, stat needs to be serialized.